src/test/java/com/findmyflight/findmyflight/service/flightwatcher/FlightWatcherControllerTest.java

```diff
@@ -0,0 +1,226 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.MediaType;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Collection;
+
+class FlightWatcherControllerTest extends IntegrationTest {
+
+    @Autowired
+    private FlightWatcherMapper flightWatcherMapper;
+
+    @AfterEach
+    void cleanUp() {
+        flightWatcherCreator.deleteAll();
+    }
+
+    @Nested
+    class FindAllTest {
+        @Test
+        void givenFlightWatchers_thenReturnAll() {
+            // given
+            FlightWatcher sample1 = flightWatcherCreator.createSample();
+            FlightWatcher sample2 = flightWatcherCreator.createSample();
+            FlightWatcher sample3 = flightWatcherCreator.createSample();
+            // when
+            // then
+            Collection<FlightWatcherResponse> response = webTestClient.get()
+                    .uri("/flight-watcher")
+                    .exchange()
+                    .expectStatus().isOk()
+                    .expectBody(new ParameterizedTypeReference<Collection<FlightWatcherResponse>>() {
+                    })
+                    .returnResult()
+                    .getResponseBody();
+
+            Assertions.assertThat(response)
+                    .containsExactly(
+                            flightWatcherMapper.map(sample1),
+                            flightWatcherMapper.map(sample2),
+                            flightWatcherMapper.map(sample3)
+                    );
+        }
+    }
+
+    @Nested
+    class FindByIdTest {
+        @Test
+        void givenFlightWatcher_thenReturn() {
+            // given
+            FlightWatcher sample = flightWatcherCreator.createSample();
+            // when
+            // then
+            webTestClient.get()
+                    .uri("/flight-watcher/{id}", sample.getId())
+                    .exchange()
+                    .expectStatus().isOk()
+                    .expectBody()
+                    .jsonPath("$.id").isEqualTo(sample.getId())
+                    .jsonPath("$.fromCity").isEqualTo(sample.getFromCity())
+                    .jsonPath("$.toCity").isEqualTo(sample.getToCity())
+                    .jsonPath("$.fromDate").isEqualTo(sample.getFromDate().format(DateTimeFormatter.ISO_DATE))
+                    .jsonPath("$.toDate").isEqualTo(sample.getToDate().format(DateTimeFormatter.ISO_DATE))
+                    .jsonPath("$.suspended").isEqualTo(sample.getSuspended());
+        }
+    }
+
+    @Nested
+    class CreateTest {
+        @Test
+        void givenValidRequest_thenCreate() {
+            //given
+            EmailNotificationReceiver sampleReceiver = emailNotificationReceiverCreator.createSample();
+            CreateOrUpdateFlightWatcherRequest request = CreateOrUpdateFlightWatcherRequest.builder()
```

Comment:
This builder may be a private class or not ?

Reply:
A builder cannot be a private class, because we want to use it outside of a given class' scope to build this class' instances. So this builder have to be public.
Of course, you can use [builder pattern](https://refactoring.guru/design-patterns/builder) in private scope - it's completely valid, if it's use-case stays "in private scope". But here, and for most use-cases of Lombok builders, we want to make it easier to build inscances wherever needed, so it's public.